Shared/DsaController.h

```diff
@@ -61,6 +64,7 @@ private slots:
   void errorOccurred(const QString& message, const QString& additionalMessage);
 
 private:
+  static const DsaController* s_instance;
```

Comment:
```suggestion
  static inline const DsaController* s_instance = nullptr;
```

'inline' in this context means single definition so you don't need do add any initialization code to the .cpp file for it.